src/components/BetterSwipeable.tsx

```diff
@@ -0,0 +1,440 @@
+   */
+  overshootRight?: boolean;
+
+  /**
+   * Specifies how much the visual interaction will be delayed compared to the
+   * gesture distance at overshoot. Default value is 1, it mean no friction, for
+   * a native feel, try 8 or above.
+   */
+  overshootFriction?: number;
+
+  /**
+   * Called when left action panel gets open.
+   */
+  onSwipeableLeftOpen?: () => void;
+
+  /**
+   * Called when right action panel gets open.
+   */
+  onSwipeableRightOpen?: () => void;
+
+  /**
+   * Called when action panel gets open (either right or left).
+   */
+  onSwipeableOpen?: () => void;
+
+  /**
+   * Called when action panel is closed.
+   */
+  onSwipeableClose?: () => void;
+
+  /**
+   * Called when left action panel starts animating on open.
+   */
+  onSwipeableLeftWillOpen?: () => void;
+
+  /**
+   * Called when right action panel starts animating on open.
+   */
+  onSwipeableRightWillOpen?: () => void;
+
+  /**
+   * Called when action panel starts animating on open (either right or left).
+   */
+  onSwipeableWillOpen?: () => void;
+
+  /**
+   * Called when action panel starts animating on close.
+   */
+  onSwipeableWillClose?: () => void;
+
+  /**
+   *
+   * This map describes the values to use as inputRange for extra interpolation:
+   * AnimatedValue: [startValue, endValue]
+   *
+   * progressAnimatedValue: [0, 1] dragAnimatedValue: [0, +]
+   *
+   * To support `rtl` flexbox layouts use `flexDirection` styling.
+   * */
+  renderLeftActions?: (
+    progressValue: Animated.SharedValue<number>,
+    dragValue: Animated.SharedValue<number>
+  ) => React.ReactNode;
+  /**
+   *
+   * This map describes the values to use as inputRange for extra interpolation:
+   * AnimatedValue: [startValue, endValue]
+   *
+   * progressAnimatedValue: [0, 1] dragAnimatedValue: [0, -]
+   *
+   * To support `rtl` flexbox layouts use `flexDirection` styling.
+   * */
+  renderRightActions?: (
+    progressValue: Animated.SharedValue<number>,
+    dragValue: Animated.SharedValue<number>
+  ) => React.ReactNode;
+
+  animationOptions?: Animated.WithSpringConfig;
+
+  /**
+   * Style object for the container (`Animated.View`), for example to override
+   * `overflow: 'hidden'`.
+   */
+  containerStyle?: StyleProp<ViewStyle>;
+
+  /**
+   * Style object for the children container (`Animated.View`), for example to
+   * apply `flex: 1`
+   */
+  childrenContainerStyle?: StyleProp<ViewStyle>;
+
+  // typescript is determined that component created with forwardRef cannot have children
+  children?: React.ReactNode;
+}
+
+export interface SwipeableController {
+  close: () => void;
+  openLeft: () => void;
+  openRight: () => void;
+  panGesture: PanGesture;
+}
+
+export const Swipeable = React.forwardRef<SwipeableController, SwipeableProps>(
+  (props, ref) => {
+    const { children, renderLeftActions, renderRightActions } = props;
+
+    const [rowState, setRowState] = useState(0); // current state: 1 - left, 0 - closed, -1 - right
```

Comment:
The state should be a const enum with strings.

Reply:
Done in 22e4c08bc7201b3e1d936ae2f01abf5d77f48261